### PR TITLE
PFRecHit Producer for HGCal Hits

### PR DESCRIPTION
**Pull request by lgray**
Implement dummy navigation module while we are making the real thing.
Import PFRecHit creation modules for HGCal to 80X from 62X.

This is a technical PR, no changes expected observed.

### CODE REVIEW

**Comment by lgray** [comment]
@cmsbuild please test

File: RecoParticleFlow/PFClusterProducer/interface/HGCRecHitNavigator.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          ForwardSubdetector D3, typename hgcheb>
+class HGCRecHitNavigator : public PFRecHitNavigatorBase {
+ public:
+  HGCRecHitNavigator() {
+  }
+
+
+
+  HGCRecHitNavigator(const edm::ParameterSet& iConfig) {
+    if( iConfig.exists("hgcee") ) {
```
Comment (slava77) [inline RecoParticleFlow/PFClusterProducer/interface/HGCRecHitNavigator.h]: please use fillDescription
(or explain why it's not possible)

File: RecoParticleFlow/PFClusterProducer/interface/PFRecHitQTests.h [hunk 2]
```diff
@@ -539,10 +566,83 @@ class PFRecHitQTestHCALCalib29 : public PFRecHitQTestBase {
...
+      throw cms::Exception("WrongDetector")
+	<< "PFRecHitQTestThresholdInMIPs only works for HGCAL!";
+      return false;
+    }
+
+    bool test(reco::PFRecHit& hit,const HGCRecHit& rh,bool& clean) {
+      const double newE = ( recHitEnergy_keV_ ? 
+			    1.0e-6*rh.energy()*recHitEnergyMultiplier_ :
+			    rh.energy()*recHitEnergyMultiplier_ );
+      /*
```
Comment (slava77) [inline RecoParticleFlow/PFClusterProducer/interface/PFRecHitQTests.h]: here and below, please remove the commented out code.
If it's needed, change to LogDebug or LogTrace

**Comment by cvuosalo** [comment]
@lgray: Could you please explain how to test this new module? What config can be used with non-empty inputs?

**Comment by lgray** [comment]
@cvuosalo The module is automatically tested in RecoLocalCalo/HGCalRecProducers/test/testHGCalRecoLocal_cfg.py which is run as a unit test (for now).

**Comment by cvuosalo** [comment]
@lgray: What part of the new code is the "dummy module" and what part will be permanent? I'm wondering what parts to review and whether to be concerned about the use of bare pointers, for example.

**Comment by lgray** [comment]
Anything related to navigation from rechit to rechit for HGC is going to be replaced at some point.

**Comment by lgray** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

For #12947 2312023e07852aa5701807975b58aa9d6755abf3

Preparing an HGCal PF rec hit navigation module for later development. There should be no change in monitored quantities.

The code changes are satisfactory, and Jenkins [tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12947/10534/summary.html) against baseline CMSSW_8_0_X_2016-01-14-2300 show no significant differences, as expected. The new module runs successfully in the Jenkins unit test.

**Comment by davidlange6** [comment]
+1

**Pull request merged**